pallets/creditcoin/src/lib.rs

```diff
@@ -1417,6 +1508,7 @@ pub mod pallet {
 			ensure!(!Currencies::<T>::contains_key(&id), Error::<T>::CurrencyAlreadyRegistered);
 
 			Currencies::<T>::insert(&id, &currency);
+			Self::deposit_event(Event::<T>::CurrencyRegistered(id, currency));
```

Comment:
I'm not seeing an explicit assertion for this event. See my other comment about `register_currency()`